### ISSUE TITLE
cannot create a FlowVar[ tuple[a: string] ]

### ISSUE THREAD

**Issue by rayman22201** [nim-lang/Nim#11717]
Attempting to create a `FlowVar[ tuple[ a: string ] ]` fails with:
`test.nim(6, 15) Error: cannot create a flowVar of type: tuple[a: string]`

Creating a `FlowVar[string]` directly works.
Creating a `FlowVar` of a `tuple` with fields of any other type works. Only tuples containing strings seem to fail.

This was discovered by attempting to use `execCmdEx()` from the `OsProc` module with `spawn`.

### Example
```nim
import threadpool, osproc

proc test(): tuple[a: string] =
  return (a: "flowvar string tuple")

let a = spawn test()

sync()

echo ^a
```

```
$ nim -v
Nim Compiler Version 0.20.99 [Linux: amd64]
Compiled at 2019-07-11
Copyright (c) 2006-2019 by Andreas Rumpf

active boot switches: -d:release
```

**Comment by alaviss** [comment]
it's worth noting that this works with https://github.com/yglukhov/threadpools :)

**Comment by andrewbriggs** [comment]
@rayman22201 Looking forward to seeing you again when I'm back in Vegas buddy!

**Comment by Chris3606** [comment]
This still shows up in Nim v1.4.2; is there an intended workaround for this, short of using the linked version of threadpools?

**Comment by alaviss** [comment]
Nope, official threadpool is sorta unmaintained by now. Alternatives includes https://github.com/mratsim/weave and https://github.com/yglukhov/threadpools

**Comment by Chris3606** [comment]
Hm.  I can definitely work around it or use an alternative for my use case...  But for future reference (both for me and those who encounter this issue in the future), given that, are there future plans to remove/replace the current Nim standard library threadpool implementation?  Particularly since one of the linked implementations appears to not support `gc:arc`...

**Comment by alaviss** [comment]
AFAIK weave will be the future. ARC supports will come after we got `Isolated[T]` in place I think, since it would be much easier to move memory between threads. See https://github.com/nim-lang/RFCs/issues/160 and https://github.com/nim-lang/RFCs/issues/244 and https://github.com/nim-lang/RFCs/issues/295 for the future of Nim's multithreading/async.

**Comment by ringabout** [comment]
It seems to work with `--gc:arc`?

```nim
import threadpool

proc test(): tuple[a: string] =
  return (a: "flowvar string tuple")

let a = spawn test()

sync()

echo typeof(a)
echo ^a
```

```
FlowVar[tuple[a: string]]
(a: "flowvar string tuple")
```
